x/batch/decorator_test.go

```diff
@@ -89,190 +116,208 @@ func mockLog(num int64, content string) string {
 	return strings.Join(list, "\n")
 }
 
-func TestDecorator(t *testing.T) {
-	Convey("Test Decorator", t, func() {
-		msg := &mockMsg{}
-		helper := &mockHelper{}
-		decorator := batch.NewDecorator()
-		Convey("Happy path", func() {
-			num := int64(10)
-			logVal := "log"
-			dataContent := make([]byte, 1)
-			gas := int64(1)
-			fee := coin.Coin{Whole: 1, Fractional: 400000000, Ticker: "IOV"}
-			combinedFee, err := fee.Multiply(num)
-			So(err, ShouldBeNil)
-
-			msg.On("Validate").Return(nil).Times(2)
-			msg.On("MsgList").Return(make([]weave.Msg, num), nil).Times(2)
-			helper.On("GetMsg").Return(msg, nil).Times(2)
-
-			helper.On("Check", nil, nil, mock.Anything).Return(&weave.CheckResult{
-				Data:         make([]byte, 1),
-				Log:          logVal,
-				GasAllocated: gas,
-				GasPayment:   gas,
-				RequiredFee:  fee,
-			}, nil).Times(int(num))
-
-			checkRes, err := decorator.Check(nil, nil, helper, helper)
-			So(err, ShouldBeNil)
-			data, _ := mockData(num, dataContent).Marshal()
-			So(checkRes, ShouldResemble, &weave.CheckResult{
-				Data:         data,
-				Log:          mockLog(num, logVal),
-				GasAllocated: gas * num,
-				GasPayment:   gas * num,
-				RequiredFee:  combinedFee,
-			})
-
-			helper.On("Deliver", nil, nil, mock.Anything).Return(&weave.DeliverResult{
-				Data:        make([]byte, 1),
-				Log:         logVal,
-				GasUsed:     gas,
-				Diff:        make([]weave.ValidatorUpdate, 1),
-				Tags:        make([]common.KVPair, 1),
-				RequiredFee: fee,
-			}, nil).Times(int(num))
-
-			deliverRes, err := decorator.Deliver(nil, nil, helper, helper)
-			So(err, ShouldBeNil)
-			So(deliverRes, ShouldResemble, &weave.DeliverResult{
-				Data:        data,
-				Log:         mockLog(num, logVal),
-				GasUsed:     gas * num,
-				Diff:        mockDiff(num),
-				Tags:        mockTags(num),
-				RequiredFee: combinedFee,
-			})
-			helper.AssertExpectations(t)
-			msg.AssertExpectations(t)
-		})
+func TestDecoratorNew(t *testing.T) {
+	logVal := "log"
+	gas := int64(1)
+	data := make([]byte, 1)
+	makeRes := func(req coin.Coin) *weave.DeliverResult {
+		return &weave.DeliverResult{
+			Data:        data,
+			Log:         logVal,
+			GasUsed:     gas,
+			Diff:        make([]weave.ValidatorUpdate, 1),
+			Tags:        make([]common.KVPair, 1),
```

Comment:
Both `Diff` and `Tags` are initialized with a slice of size 1 with no elements in them. Is this on purpose?

Reply:
Yes, this was in the original test as it does not make any difference.